### PR TITLE
Add 4 capability tasks for screening matrix; run results in description

### PR DESCRIPTION
**Pull request by amc-corey-cox**
## Summary

Adds 4 new eval tasks designed to give meaningful capability differentiation across local models, plus the **screening matrix results** from running these (and the 2 pre-existing tasks) across 3 models. Results table is also committed to `docs/CLAUDE-CODE-LOCAL-MODELS.md` as part of this PR so it persists post-merge.

## New tasks

| Task | What it tests |
|------|---------------|
| `03-add-docstring` | Locate a function, insert a docstring in the right place, preserve syntax |
| `04-rename-variable` | Consistent symbol rename across all uses in a file, no over-matching |
| `05-fix-failing-test` | Read failing assertion output, localize bug to the right file, fix |
| `06-implement-from-stub` | Read a stub + tests, derive what to implement, handle edge cases |

Each follows the existing `prompt.txt` + `setup.sh` + `score.sh` + `expect.md` contract from #62. All `score.sh` checks are deterministic.

## Local pre-validation

Before running the matrix, I applied the canonical "correct fix" by hand for each new task and ran `score.sh`:

```
03-add-docstring: PASS
04-rename-variable: PASS
05-fix-failing-test: PASS
06-implement-from-stub: PASS
```

So task design is verified separately from model performance.

## Models in the matrix

Chosen by probing `ollama show <model> | grep tools` on what was already pulled on the eval host. Tool-calling support is required — Claude Code's harness depends on it. Three candidates passed: `qwen3-coder-32k`, `gpt-oss:20b`, `gpt-oss-64k`. Two pulled models were rejected at this gate (see "Bonus discovery" below).

## Screening matrix results

**Hardware:** server CPU only

| Task | qwen3-coder-32k | gpt-oss:20b | gpt-oss-64k |
|------|-----------------|-------------|-------------|
| 01-sanity-pong | ran (8 min) | ran (2 min) | timed out (58 min) |
| 02-fix-typo | **PASS** (28 min) | fail (7 min) | timed out (58 min) |
| 03-add-docstring | **PASS** (28 min) | fail (96 min) | timed out (58 min) |
| 04-rename-variable | **PASS** (29 min) | fail (73 min) | timed out (58 min) |
| 05-fix-failing-test | **PASS** (31 min) | fail (96 min) | timed out (58 min) |
| 06-implement-from-stub | **PASS** (36 min) | fail (63 min) | **PASS** (16 min) |
| **Capability pass-rate** | **5/5** | **0/5** | **1/5** |

**Headline:** `qwen3-coder-32k` is the only model in this matrix that meaningfully drives Claude Code on this hardware. Recommendation stands. The other two would need either harness adjustments (e.g., longer Claude Code timeouts for `gpt-oss-64k`) or faster compute (Track 3, #54) to be re-evaluable.

Full interpretation, caveats, and the "I'm sorry but I can't modify the code" oddity from `gpt-oss-64k`'s only pass: see `docs/CLAUDE-CODE-LOCAL-MODELS.md` in this PR.

## Two harness bugs found and fixed mid-run

1. **Path-sanitization collisions** — the initial fix `tr ':/' '__'` mapped both `:` and `/` to `_`, which collides for some Ollama name patterns. Tightened to `tr ':/' '_-'` so they get distinct replacements.
2. **Warmup runaway** — the bare `"OK"` warmup prompt sent `qwen3-coder-32k` into a 19-turn, 3-hour agent loop before timing out. Fixed in `run-matrix.sh` with an explicit prompt + 30-min `timeout` wrapper.

The `qwen3-coder-32k` task results from before the warmup-runaway happened were still valid — those tasks ran fine; the warmup was just noisy. Data preserved by carrying that directory over to the second run rather than redoing 6 hours of work.

## Bonus discovery

Two pulled models can't be used in this harness at all and were filtered out before the matrix ran:

- `qwen:72b` — Ollama returns `"does not support tools"` immediately. Pre-tool-calling-era model.
- `vanilj/midnight-miqu-70b-v1.5` — same.

Future matrix runs should always probe `ollama show <model> | grep tools` before committing to the model list. This filter saved ~12 hours of wasted run time.

## Test plan

- [x] `bash -n` on all new scripts
- [x] Local pre-validation with correct-fix applied (all 4 new tasks PASS)
- [x] Screening matrix run on tesseract (3 models × 6 tasks)
- [x] Results table added to PR description and committed to `docs/CLAUDE-CODE-LOCAL-MODELS.md` in this PR
- [x] Review fixes re-validated: 04 import works, 05 bypass blocked, 06 still passes

### CODE REVIEW

File: scripts/eval/tasks/04-rename-variable/score.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Pass if:
+#   - app.py still imports cleanly
+#   - no standalone `count` identifier remains
+#   - `total_count` appears at least 4 times (one decl + uses)
+set -e
+
+[ -f app.py ] || { echo "app.py missing"; exit 2; }
+
+# Syntax check
+python3 -c "import ast; ast.parse(open('app.py').read())"
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/04-rename-variable/score.sh]: Fixed in 34b000e — replaced `ast.parse` with `importlib.util` module load so a top-level `raise` or `1/0` would fail scoring as expected.

File: scripts/eval/tasks/05-fix-failing-test/setup.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -e
+
+git init -q
+git config user.email "eval@forge.local"
+git config user.name "Eval Setup"
+
+cat > calc.py <<'PY'
+def add(a, b):
+    return a - b  # bug: should be +
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/05-fix-failing-test/setup.sh]: Fixed in 34b000e — removed the inline `# bug: should be +` comment. The model now has to localize the bug from the failing assertion output rather than a hint in source.

File: scripts/eval/tasks/05-fix-failing-test/score.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+set -e
+
+[ -f calc.py ] || { echo "calc.py missing"; exit 2; }
+[ -f check.py ] || { echo "check.py missing"; exit 2; }
+
+# Confirm the model didn't sidestep by editing check.py
+if ! git -C . diff HEAD --exit-code -- check.py > /dev/null 2>&1; then
+    echo "check.py was modified — the fix should be in calc.py only"
+    exit 3
+fi
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/05-fix-failing-test/score.sh]: Fixed in 34b000e — diff now goes against the root commit (`git rev-list --max-parents=0 HEAD`) instead of HEAD, so a commit by the agent can't hide the edit. Verified locally: applying a check.py edit + commit is correctly rejected by score.sh.

[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/05-fix-failing-test/score.sh]: Fixed in 34b000e — same root-commit diff as the comment on this line; verified the commit-after-edit bypass is now blocked.

File: scripts/eval/tasks/06-implement-from-stub/score.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# to check.py — the implementation has to live in palindrome.py.
+set -e
+
+[ -f palindrome.py ] || { echo "palindrome.py missing"; exit 2; }
+[ -f check.py ] || { echo "check.py missing"; exit 2; }
+
+if ! git -C . diff HEAD --exit-code -- check.py > /dev/null 2>&1; then
+    echo "check.py was modified — the implementation should be in palindrome.py only"
+    exit 3
+fi
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/06-implement-from-stub/score.sh]: Fixed in 34b000e — root-commit diff for 06 same as 05.

[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/tasks/06-implement-from-stub/score.sh:13]: Fixed in 34b000e — same root-commit fix as 05.

File: scripts/eval/run-task.sh [hunk 5]
```diff
@@ -42,7 +42,12 @@ PROMPT_FILE="$TASK_DIR/prompt.txt"
 TASK_NAME=$(basename "$TASK_DIR")
 TASK_DIR_ABS=$(cd "$TASK_DIR" && pwd)
 
-OUT="$OUTPUT_BASE/$MODEL/$TASK_NAME"
+# Sanitize model name for filesystem use. Ollama names like "qwen:72b" contain
+# colons that break docker bind-mount path parsing if used directly. The original
+# name still gets passed to `claude -p --model` and recorded in meta.json.
+MODEL_PATH=$(echo "$MODEL" | tr ':/' '__')
+
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/run-task.sh:51]: Fixed in 34b000e — changed `tr` mapping from `':/' '__'` to `':/' '_-'` so `:` and `/` get distinct replacements; collisions between names like `org/model:tag` and `org_model/tag` no longer possible.

File: scripts/eval/run-matrix.sh [hunk 6]
```diff
@@ -35,30 +35,42 @@ echo "  tasks_dir: $TASKS_DIR"
...
+WARMUP_PROMPT="Reply with the single word OK and nothing else."
+WARMUP_TIMEOUT_S=1800
+
 warmup() {
     local model="$1"
-    local warmup_out="$OUTPUT_BASE/_warmup/$model"
+    local warmup_path
+    warmup_path=$(echo "$model" | tr ':/' '__')
+    local warmup_out="$OUTPUT_BASE/_warmup/$warmup_path"
     mkdir -p "$warmup_out"
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply scripts/eval/run-matrix.sh:52]: Fixed in 34b000e — same `tr ":/" "_-"` change for the warmup_path.

File: docs/CLAUDE-CODE-LOCAL-MODELS.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 How Claude Code performs against local Ollama models, and which ones we recommend. Part of the [Track 1 evaluation](https://github.com/amc-corey-cox/cc_forge/issues) of the agent-architecture exploration.
 
-> **Status:** Initial cost-model captured (see below). Matrix results pending the first evaluation pass.
+> **Status:** First screening matrix run complete (2026-06-17). Headline result: `qwen3-coder-32k` is the only model in our initial shortlist that meaningfully drives Claude Code in this harness. Recommendation stands as the dogfood default.
```
[the remark this answers is not in the document]
Comment (amc-corey-cox) [reply docs/CLAUDE-CODE-LOCAL-MODELS.md:5]: Updated PR description to reorganize: the matrix shortlist (`qwen3-coder-32k`, `gpt-oss:20b`, `gpt-oss-64k`) is now clearly distinguished from the "Bonus discovery" section about `qwen:72b` and `vanilj/midnight-miqu-70b-v1.5`, which were filtered OUT before the matrix ran because they don't support tool calling. The doc and the PR description now refer to the same three models in the matrix; no contradiction. The doc is committed in this PR (as `c6df054`), not a follow-up — the description now states that explicitly.
